cloudant-client/src/test/java/com/cloudant/tests/AttachmentsTest.java

```diff
@@ -241,26 +221,42 @@ public void attachmentStandaloneGivenId() throws IOException, URISyntaxException
         }
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void attachmentStandaloneNullDocNonNullRev() {
-        byte[] bytesToDB = "binary data".getBytes();
-        ByteArrayInputStream bytesIn = new ByteArrayInputStream(bytesToDB);
-        Response response = db.saveAttachment(bytesIn, "foo.txt", "text/plain", null, "1-abcdef");
+        assertThrows(IllegalArgumentException.class, new Executable() {
```

Comment:
There are 49 instances of `assertThrows` and wrapping the whole thing in an `Executable` was easier, because:
- That was what the IDE auto-prompted for (lazy, I know)
- It's difficult to tell with each test which statement is the exact one responsible for throwing the exception, which i guess is one of the drawbacks of the old `@Throws` annotation.

So it takes time to understand each test and put the `assertThrows` in the right place, but I agree that the way you have suggested doing it is a lot clearer and more expressive.

Possibly something to look at towards the end of the PR?